packages/trader/src/Modules/CFD/Components/cfd-personal-details-form.tsx

```diff
@@ -40,64 +179,76 @@ const getAccountOpeningReasonList = () => [
                 type='text'
                 required={!optional}
                 name={name}
                 autoComplete='off'
                 maxLength={maxLength || '30'}
                 error={touched[field.name] && errors[field.name]}
-                {...field}
                 {...props}
             />
         )}
     </Field>
 );
 
-const validatePersonalDetails = ({ values, residence_list, account_opening_reason, is_tin_required }) => {
+const validatePersonalDetails = ({
+    values,
+    residence_list,
+    account_opening_reason,
+    is_tin_required,
+}: TValidatePersonalDetailsParams) => {
     const [tax_residence_obj] = residence_list.filter(res => res.text === values.tax_residence && res.tin_format);
     const [tin_format] = tax_residence_obj?.tin_format ?? [];
     const tin_regex = tin_format || '^[A-Za-z0-9./s-]{0,25}$'; // fallback to API's default rule check
 
     const validations = {
-        citizen: [v => !!v, v => residence_list.map(i => i.text).includes(v)],
-        tax_residence: [v => !!v, v => residence_list.map(i => i.text).includes(v)],
+        citizen: [(v: string) => !!v, (v: string) => residence_list.map(i => i.text).includes(v)],
+        tax_residence: [(v: string) => !!v, (v: string) => residence_list.map(i => i.text).includes(v)],
         tax_identification_number: [
-            v => ((!values.tax_residence && is_tin_required) || tin_format ? !!v : true),
-            v => (tin_regex ? v.match(tin_regex) : true),
+            (v: string) => ((!values.tax_residence && is_tin_required) || tin_format ? !!v : true),
+            (v: string) => (tin_regex ? v.match(tin_regex) : true),
+        ],
+        account_opening_reason: [
+            (v: string) => !!v,
+            (v: string) => account_opening_reason.map(i => i.value).includes(v),
         ],
-        account_opening_reason: [v => !!v, v => account_opening_reason.map(i => i.value).includes(v)],
     };
-    const mappedKey = {
+    const mappedKey: { [key: string]: string } = {
         citizen: localize('Citizenship'),
         tax_residence: localize('Tax residence'),
         tax_identification_number: localize('Tax identification number'),
         account_opening_reason: localize('Account opening reason'),
     };
 
-    const field_error_messages = field_name => [
+    const field_error_messages = (field_name: string): string[] => [
         localize('{{field_name}} is required', { field_name }),
         localize('{{field_name}} is not properly formatted.', { field_name }),
     ];
 
-    const errors = {};
+    const errors: { [key: string]: React.ReactNode } = {};
 
     Object.entries(validations).forEach(([key, rules]) => {
-        const error_index = rules.findIndex(v => !v(values[key]));
+        const error_index = rules.findIndex(v => !v(values[key as keyof TFormValues]));
         if (error_index !== -1) {
-            errors[key] = <React.Fragment>{field_error_messages(mappedKey[key])[error_index]}</React.Fragment>;
```

Comment:
I have to get rid of <React.Fragment> here because:
1) None of the components, which we pass the resulting error further to, needs this error to be a node;
2) The error should be a string because Input is expecting a string at line 192 otherwise we get a console error.